### DISCUSSION TITLE
Ideas to update the cached list of projects

### DISCUSSION THREAD

**Discussion by rkulla** [cljoly/telescope-git-repo.nvim#64] [Ideas]
I see there's a TODO item in the README for periodic updating of cache_list. Using your gupdatedb advice from `brew install findutils`, this cron job worked for me to update hourly:

    0 * * * * /usr/local/bin/gupdatedb --localpaths=$HOME --prunepaths="/Volumes .*node_modules.*" --output="$HOME/locatedb" > "$HOME/cron-output-loaddb.log" 2>&1

however, is there a reason you recommend gnu locate and updatedb, when equivalent bsd commands are already installed on (most) macs?  There's also `mdfind` which taps into Spotlight's db (but is supposedly slower than locate). 

Not to mention that it's more common on MacOS to use launchctl a .plist file?

All of this automation requires some additional permissions for the scripts to generate files, thanks to SIP. So if someone understands this all better than me, they should consider coming up with a solution that is the least confusing and has troubleshooting steps, such as `putil -lint /path/to/.plist`, `log show`, etc.

Otherwise it's possibly not worth adding this todo item other than to say something generic like "Use your favorite scheduler" (and leave it up to the person to figure out).

**Comment by rkulla** [comment]
Forgot to mention that in the above cron `/usr/local/bin/gupdatedb` is for intel mac's but could be `/opt/homebrew/bin/gupdatedb` on an m-chip (apple-silicon) Mac. People would have to consult `which gupdatedb` and update it accordingly and doing anything dynamically in cron is a bit hard to do because of its locked down shell environment.

**Comment by rkulla** [comment]
I personally also prefer the cached_list to load extremely fast, which requires keeping locatedb much smaller.  To do this, I have all my important repos for work/home projects within just `$HOME/repos`, e.g., `$HOME/repos/work  $HOME/repos/personal` and only index that:

    # Keeping locatedb small (by only loading my $HOME/repos path and ignoring node_modules, makes it load MUCH faster.
    # I use `mfind` on the cli to use spotlights more extensive database
    alias loaddb="gupdatedb --localpaths=$HOME/repos --prunepaths='.*node_modules.*' --output=$HOME/locatedb"

This keeps my ~/locatedb around 1-2MB instead of 20+ MB and loads instantly from telescope. It also generates the db so fast, that I can run loaddb in less than 2 seconds. So I can more frequently run the cron job to keep it up-to-date: 

    0/5 * * * * /usr/local/bin/gupdatedb --localpaths=$HOME/repos --prunepaths=".*node_modules.*" --output="$HOME/locatedb" > "$HOME/cron-output-loaddb.log" 2>&1

**Comment by cljoly** [comment]
Thanks for sharing @rkulla!

> however, is there a reason you recommend gnu locate and updatedb, when equivalent bsd commands are already installed on (most) macs?

Yes, I don’t have a mac for personal use, so it’s hard to debug the inevitable corner cases that later cause issues :).

I’m happy to update the README with something more generic and to redirect to a GitHub discussion thread for more detailed discussions.

**Reply by rkulla** [reply]
@cljoly sounds good. thanks!

**Reply by cljoly** [reply]
Done, [PR opened](#65), I’ll wait a bit to merge it in case you wish to comment on it.

**Comment by cljoly** [comment]
This was also discussed in https://github.com/cljoly/telescope-repo.nvim/discussions/13
